File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/shared/videoeditortokenresponse.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class VideoEditorTokenResponseToken:
    r"""VideoEditorTokenResponseToken
    The user access token
    """
    
    expiration_date: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('expirationDate') }})
    value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    

@dataclass_json
@dataclasses.dataclass
class VideoEditorTokenResponse:
    r"""VideoEditorTokenResponse
    The response that includes the user access token and its expiration date
    """
    
    token: VideoEditorTokenResponseToken = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('token') }})
    

File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/shared/videoeditortokenresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// VideoEditorTokenResponseToken
/** 
 * The user access token
**/
export class VideoEditorTokenResponseToken extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=expirationDate" })
  expirationDate?: number;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value?: string;
}


// VideoEditorTokenResponse
/** 
 * The response that includes the user access token and its expiration date
**/
export class VideoEditorTokenResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=token" })
  token: VideoEditorTokenResponseToken;
}
